File: tps/gabarito/tp01/TP01-Palindromo-C/palindromo.c

/**
 * @author: Thiago de Campos Ribeiro Nolasco
*/

// Bibliotecas utilizadas:
#include <stdio.h>
#include <string.h>
#include <locale.h>

#define true 1
#define false 0

// Prototipação:
int isFim(char s[]);
int palindrome(char s[]);

// Receives an input from keyboard and starts the program
int main(){
	setlocale(LC_ALL,"");
	// data declaration
	char str[1000];

//	scanf("%[^\n]", str);
//	getchar();

	fgets(str, 1000, stdin);
	setbuf(stdin, NULL);
	if(str[strlen(str) - 1] == '\n'){
		str[strlen(str) - 1] = '\0';
	}
	while(!(isFim(str))){
		
		if(palindrome(str)){
			printf("SIM\n");
		}else{
			printf("NAO\n");
		}
	
		fgets(str, 1000, stdin);
		setbuf(stdin, NULL);
		if(str[strlen(str) - 1] == '\n'){
			str[strlen(str) - 1] = '\0';
		}

	} 
	// Return without errors
	return 0;
}// ending main


// Checks if the received string is equals to "FIM"
int isFim(char s[]){
	return (strlen(s) == 3 && s[0] == 'F' && s[1] == 'I' && s[2] == 'M');
}// ending isFim

// Checks if the given word is Palindrome
int palindrome(char s[]){
	int j = strlen(s) - 1;

	for(int i = 0; i < strlen(s); i++, j--){
		if(s[i] != s[j]) return false;
	}
	
	return true;
}// ending isPalindrome

File: tps/gabarito/tp01/TP01-Palindromo-JAVA/Palindromo.java
/**
 * @author: Thiago de Campos Ribeiro Nolasco
*/


class Palindromo{

	/**
	 * checks if the given word is Palindrome
	 * @param String
	 * @return 
	*/
	public static String isPalindrome(String s){
		int j = s.length() - 1;

		for(int i = 0; i < s.length(); i++, j--){
			if(s.charAt(i) != s.charAt(j)) return "NAO";
		}
		return "SIM";
	}


	/**
	 * checks if the read word == "FIM"
	 * @param String
	 * @return
	*/
	public static boolean isFim(String s){
		return (s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
	}


	/**
	 * (MAIN) receives entries from keyboard and starts the program
	 * @param String[]
	*/
	public static void main (String[] args){
		// Data declaration
		String[] entries = new String[1000];
		int numEntries = 0;

		// Reading entries
		do{
			entries[numEntries] = MyIO.readLine();
		} while(!(isFim(entries[numEntries++])));
		numEntries--; // When it reads "FIM"

		// For each line, it'll print if the given word is Palindrome or not ("SIM" / "NAO)
		for(int i = 0; i < numEntries; i++){
			MyIO.println(isPalindrome(entries[i]));
		}
	}	

}// Fim class Palindromo
